File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/connection/testing/TestConnections.java
package com.lasthopesoftware.bluewater.client.connection.testing;

import com.lasthopesoftware.bluewater.client.connection.IConnectionProvider;
import com.namehillsoftware.handoff.promises.Promise;
import org.joda.time.Duration;

public interface TestConnections {
	Promise<Boolean> promiseIsConnectionPossible(IConnectionProvider connectionProvider);

	Promise<Boolean> promiseIsConnectionPossible(IConnectionProvider connectionProvider, Duration timeout);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/connection/testing/TestConnections.kt
package com.lasthopesoftware.bluewater.client.connection.testing

import com.lasthopesoftware.bluewater.client.connection.LiveServerConnection
import com.namehillsoftware.handoff.promises.Promise

interface TestConnections {
	fun promiseIsConnectionPossible(connectionProvider: LiveServerConnection): Promise<Boolean>
}
